beacon-chain/powchain/prometheus.go

```diff
@@ -0,0 +1,110 @@
+package powchain
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prysmaticlabs/prysm/shared/clientstats"
+	"sync"
+)
+
+type powchainCollector struct {
+	SyncEth1Connected *prometheus.Desc
+	SyncEth1FallbackConnected *prometheus.Desc
+	SyncEth1FallbackConfigured *prometheus.Desc // true if flag specified: --fallback-web3provider
+	updateChan chan clientstats.BeaconNodeStats
+	latestStats clientstats.BeaconNodeStats
+	sync.Mutex
+}
+
+type BeaconnodeStatsUpdater interface {
+	Update(stats clientstats.BeaconNodeStats)
+}
+
+func NewBeaconnodeStatsUpdater() (BeaconnodeStatsUpdater, error) {
```

Comment:
Any benefit to returning ifaces here instead of a concrete type?

Reply:
`BeaconnodeStatsUpdater` is an interface -- `powchainCollector` is the concrete type. Now that I look at it again, I don't think it should be exported though, so adding a todo to make it private.